### ISSUE TITLE
Detection and segmentation models

### ISSUE THREAD

**Issue by pianogGG** [NVlabs/FasterViT#21]
Hi , I saw in the catalog that you have plans to train models for detection and segmentation. I was wondering 
1. if you plan to modify the network structure for the detection task ?
2. When do you plan to release the related models? 
3. I want to replace the Swin-T backbone in BevFusion's camera backbone with FasterVit and use existing ImageNet-1K pre-trained models. Do you think this would be a good solution?

**Comment by ahatamiz** [comment]
Hi @pianogGG , yes we will release the detection code with a slightly modified architecture. Although the any-resolution FasterViT model can readily be used for this purpose at this stage as well. 

There is no ETA currently for lease of these models but hopefully that should happen soon. 

And it is certainly a great idea to use FasterViT in BEVFusion with counterpart FasterViT due to its clear advantage in both accuracy and throughput. There is some domain gap, but many papers have shown that ImageNet pre-trained models indeed prove useful.

**Comment by pianogGG** [comment]
Hi @ahatamiz ,"yes we will release the detection code with a slightly modified architecture"===> did you mean FastVit backbone or neck or head? 
![Screen Shot 2023-06-21 at 12 49 15](https://github.com/NVlabs/FasterViT/assets/31872064/f92a4a32-0bdb-4d49-9561-1991cd49fb22)

**Comment by pianogGG** [comment]
The FastVit backbone i mentioned refers to the part circle in the first box

**Comment by ahatamiz** [comment]
Thanks for the question @pianogGG . Precisely, the neck as we remove the final classification head and only extract intermediate feature maps.

**Comment by pianogGG** [comment]
Hi @ahatamiz Thanks a lot.   So "yes we will release the detection code with a slightly modified architecture", which part you changed?

**Comment by ahatamiz** [comment]
Hi @pianogGG , the detection/segmentation code is now almost same as [FasterViT_any_res model](https://github.com/NVlabs/FasterViT/blob/main/fastervit/models/faster_vit_any_res.py) which supports non-square images (as often see in COCO, ADE20K, etc.). We only have to remove the linear head (as mentioned above), also extract the output of each stage.

**Comment by ahatamiz** [comment]
Will close this issue for now.

**Comment by ahatamiz** [comment]
Thanks for your inquiry @pianogGG ! Please see our newly release [object detection repository](https://github.com/NVlabs/FasterViT/tree/main/object_detection) and the [FasterViT backbone](https://github.com/NVlabs/FasterViT/blob/main/object_detection/models/dino/fastervit.py) !

We will add pretrained checkpoints very soon !
